tests/testsuite/freshness.rs

```diff
@@ -573,7 +626,16 @@ fn rebuild_tests_if_lib_changes() {
     p.cargo("build -v").run();
     p.cargo("test -v")
         .with_status(101)
-        .with_stderr_contains("[..]cannot find function `foo`[..]")
+        .with_stderr_data(str![[r#"
+[DIRTY] foo v0.0.1 ([ROOT]/foo): the dependency foo was rebuilt ([TIME_DIFF_AFTER_LAST_BUILD])
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo [..]
+[RUNNING] `rustc --crate-name foo [..]
+error[E0425]: cannot find function `foo` in crate `foo`
+...
+[WARNING] build failed, waiting for other jobs to finish...
+
+"#]])
```

Comment:
By the way another flaky test `rebuild_tests_if_lib_changes`
```diff
thread 'freshness::rebuild_tests_if_lib_changes' panicked at tests\testsuite\freshness.rs:635:10:
---- expected: tests\testsuite\freshness.rs:625:27
++++ actual:   stderr
   1    1 | [DIRTY] foo v0.0.1 ([ROOT]/foo): the dependency foo was rebuilt ([TIME_DIFF_AFTER_LAST_BUILD])
   2    2 | [COMPILING] foo v0.0.1 ([ROOT]/foo)
   3    3 | [RUNNING] `rustc --crate-name foo [..]
   4    4 | [RUNNING] `rustc --crate-name foo [..]
   5    5 | error[E0425]: cannot find function `foo` in crate `foo`
-   6      - ...
-   7      - [WARNING] build failed, waiting for other jobs to finish...
+        6 +  --> tests/foo.rs:4:34
+        7 +   |
+        8 + 4 |                 fn test() { foo::foo(); }
+        9 +   |                                  ^^^ not found in `foo`
+       10 + 
+       11 + For more information about this error, try `rustc --explain E0425`.
+       12 + [ERROR] could not compile `foo` (test "foo") due to 1 previous error
+       13 + 
+       14 + Caused by:
+       15 +   process didn't exit successfully: `rustc --crate-name foo --edition=2015 tests/foo.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test --check-cfg cfg(docsrs) --check-cfg "cfg(feature, values())" -C metadata=5cb8c046256c0d51 -C extra-filename=-5cb8c046256c0d51 --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps --extern foo=[ROOT]/foo/target/debug/deps/libfoo-[HASH].rlib` ([EXIT_STATUS]: 1)
```
https://github.com/choznerol/cargo/actions/runs/9780417859/job/27002034359

Let me edit a better snapshot to avoid this.

Reply:
Fixed by 2083b3d8ff9180c0f7dea2201c0850ac0a281899